test/Microsoft.Framework.Configuration.Json.Test/ArrayTest.cs

```diff
@@ -229,16 +229,16 @@ public void PropertiesAreSortedByNumberOnlyFirst()
             builder.Add(jsonConfigSource, load: false);
             var config = builder.Build();
 
-            var configurationSection = config.GetConfigurationSection("setting");
-            var indexConfigurationSections = configurationSection.GetConfigurationSections().ToArray();
+            var configurationSection = config.GetSection("setting");
+            var indexConfigurationSections = configurationSection.GetChildren().ToArray();
 
             Assert.Equal(6, indexConfigurationSections.Count());
-            Assert.Equal("4", indexConfigurationSections[0].Key);
-            Assert.Equal("10", indexConfigurationSections[1].Key);
-            Assert.Equal("42", indexConfigurationSections[2].Key);
-            Assert.Equal("1text", indexConfigurationSections[3].Key);
-            Assert.Equal("bob", indexConfigurationSections[4].Key);
-            Assert.Equal("hello", indexConfigurationSections[5].Key);
+            Assert.Equal("setting:4", indexConfigurationSections[0].Key);
```

Comment:
Here it is different. `ConfigurationSection.Key` should always contain the full key so that based on it we can retrieve `ConfigurationSection.Value` from sources. If we don't want to do this, we should store the parent key or prefix. Do we want to go this way to store prefix to construct a key?